source/NetworkUtils.cpp

```diff
@@ -270,4 +270,169 @@ void NetworkUtils::broadcastStartSession(ENetHost* serverHost) {
+
+		outBuffer.writeFloat32(*player->jumpVelocity);
+		outBuffer.writeFloat32(*player->jumpInterval);
+		outBuffer.writeBool8(*player->jumpTouch);
+
+		outBuffer.writeFloat32(*player->height);
+		outBuffer.writeFloat32(*player->crouchHeight);
+
+		outBuffer.writeBool8(*player->headBobEnabled);
+		outBuffer.writeFloat32(*player->headBobAmplitude);
+		outBuffer.writeFloat32(*player->headBobFrequency);
+
+		outBuffer.writeVector3(*player->respawnPos);
+		outBuffer.writeBool8(true);
+
+		outBuffer.writeFloat32(*player->movementRestrictionX);
+		outBuffer.writeFloat32(*player->movementRestrictionZ);
+		outBuffer.writeBool8(*player->restrictedMovementEnabled);
+
+		outBuffer.writeBool8(*player->counterStrafing);
+	}
+	else {
+		int index = (*player->selectedClient).compare("Client 1") ? index = 0 : index = 1;
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].moveRate);
+		outBuffer.writeVector2((*player->clientPlayerConfigs)[index].moveScale);
+
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[0]);
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[1]);
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[2]);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].accelerationEnabled);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementAcceleration);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementDeceleration);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].sprintMultiplier);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].jumpVelocity);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].jumpInterval);
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].jumpTouch);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].height);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].crouchHeight);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].headBobEnabled);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].headBobAmplitude);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].headBobFrequency);
+
+		outBuffer.writeVector3((*player->clientPlayerConfigs)[index].respawnPos);
+		outBuffer.writeBool8(true);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementRestrictionX);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementRestrictionZ);
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].restrictedMovementEnabled);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].counterStrafing);
+
+		*player->readFromFile = false;
+	}
+
+	ENetPacket* packet = enet_packet_create((void*)outBuffer.getCArray(), outBuffer.length(), ENET_PACKET_FLAG_RELIABLE);
+	enet_host_broadcast(serverHost, 0, packet);
+	return 0;
+}
+
+int NetworkUtils::sendPlayerConfigToSelectedClient(ENetPeer* clientPeer, const shared_ptr<PlayerEntity>& player) {
```

Comment:
Instead of duplicating all the code above, why not have a `playerConfigToPacket()` method that creates the `BinaryOutput` here then just call different transfer methods at the end? This keeps the code more maintainable, requires fixes in fewer places.

Reply:
Thank you for your suggestion. Updated.